evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -42,8 +42,9 @@ class MockEstimator(Estimator):
         supported_problem_types = [ProblemTypes.BINARY, ProblemTypes.MULTICLASS]
         hyperparameter_ranges = {'dummy_parameter': ['default', 'other']}
 
-        def __init__(self, dummy_parameter='default', random_state=0):
-            super().__init__(parameters={'dummy_parameter': dummy_parameter}, component_obj=None, random_state=random_state)
+        def __init__(self, dummy_parameter='default', random_state=0, **kwargs):
+            super().__init__(parameters={'dummy_parameter': dummy_parameter, **kwargs},
```

Comment:
I need this to accept kwargs for one of my tests but we should do this anyway because our convention is to allow kwargs to estimators.